pkg/controller/clusterpool/clusterpool_controller.go

```diff
@@ -902,76 +858,3 @@ func (r *ReconcileClusterPool) createCloudBuilder(pool *hivev1.ClusterPool, logg
 		return nil, errors.New("unsupported platform")
 	}
 }
-
-// getAllPendingClusterClaims returns all of the ClusterClaims that are requesting clusters from the specified pool.
-// The claims are returned in order of creation time, from oldest to youngest.
-func (r *ReconcileClusterPool) getAllPendingClusterClaims(pool *hivev1.ClusterPool, logger log.FieldLogger) ([]*hivev1.ClusterClaim, error) {
-	claimsList := &hivev1.ClusterClaimList{}
-	if err := r.Client.List(context.Background(), claimsList, client.InNamespace(pool.Namespace)); err != nil {
-		logger.WithError(err).Error("error listing ClusterClaims")
-		return nil, err
-	}
-	var pendingClaims []*hivev1.ClusterClaim
-	for i, claim := range claimsList.Items {
-		// skip claims for other pools
-		if claim.Spec.ClusterPoolName != pool.Name {
-			continue
-		}
-		// skip claims that have been assigned already
-		if claim.Spec.Namespace != "" {
-			continue
-		}
-		pendingClaims = append(pendingClaims, &claimsList.Items[i])
-	}
-	sort.Slice(
-		pendingClaims,
-		func(i, j int) bool {
-			return pendingClaims[i].CreationTimestamp.Before(&pendingClaims[j].CreationTimestamp)
-		},
-	)
-	return pendingClaims, nil
-}
-
-func (r *ReconcileClusterPool) assignClustersToClaims(claims []*hivev1.ClusterClaim, cds []*hivev1.ClusterDeployment, logger log.FieldLogger) ([]*hivev1.ClusterDeployment, error) {
-	for _, claim := range claims {
-		logger := logger.WithField("claim", claim.Name)
-		var conds []hivev1.ClusterClaimCondition
-		var statusChanged bool
-		if len(cds) > 0 {
-			claim.Spec.Namespace = cds[0].Namespace
-			cds = cds[1:]
-			logger.WithField("cluster", claim.Spec.Namespace).Info("assigning cluster to claim")
-			if err := r.Update(context.Background(), claim); err != nil {
-				logger.WithError(err).Log(controllerutils.LogLevel(err), "could not assign cluster to claim")
-				return cds, err
-			}
-			conds = controllerutils.SetClusterClaimCondition(
-				claim.Status.Conditions,
-				hivev1.ClusterClaimPendingCondition,
-				corev1.ConditionTrue,
-				"ClusterAssigned",
-				"Cluster assigned to ClusterClaim, awaiting claim",
-				controllerutils.UpdateConditionIfReasonOrMessageChange,
-			)
-			statusChanged = true
-		} else {
-			logger.Debug("no clusters ready to assign to claim")
-			conds, statusChanged = controllerutils.SetClusterClaimConditionWithChangeCheck(
-				claim.Status.Conditions,
-				hivev1.ClusterClaimPendingCondition,
-				corev1.ConditionTrue,
-				"NoClusters",
-				"No clusters in pool are ready to be claimed",
-				controllerutils.UpdateConditionIfReasonOrMessageChange,
-			)
```

Comment:
this condition doesn't seem to be taken care of in the new collections?

Reply:
This is done, with the help of #1489 to validate it.